File: 525-contiguous-array/contiguous-array.cs
public class Solution {
    public int FindMaxLength(int[] nums) {
        int n = nums.Length ;
        Dictionary<int,int> map = new Dictionary<int,int>();

        map.Add(0,-1);
        int rsum = 0;
        int max = 0;
        for(int i = 0;i<n;i++){
            if(nums[i] == 1) {
                rsum = rsum+1;
            }else{
                rsum = rsum-1;
            }

            if(map.ContainsKey(rsum)){
                int diff = i - map[rsum];
                max = Math.Max(max,diff);
            }else{
                map.Add(rsum,i);
            }
        }

        return max;
    }
}

File: 525-contiguous-array/contiguous-array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        
        unordered_map<int,int> map ;
        map[0] = -1;
            int rsum = 0;
            int maxval = 0;
        for(int i = 0 ;i<nums.size();i++){
            if(nums[i] == 1){
                    rsum = rsum +1;
            }else{
                rsum = rsum-1;
            }

       if (map.find(rsum) != map.end()) { // Key exists
    int diff = i - map[rsum];
    maxval = max(maxval, diff);
} else { // Key does not exist, store its first occurrence
    map[rsum] = i;
}
            
        }

        return maxval;
    }
};